npc.cpp

```diff
@@ -62,34 +61,16 @@ void Game::GrowWMons(int rate)
 	for (level = 0; level < regions.numLevels; level++) {
 		ARegionArray *pArr = regions.pRegionArrays[level];
 
-		printf("\n\n LEVEL: %d. X: %d. Y: %d \n\n", level, pArr->x,  pArr->y);
-
-		int xsec;
-		int blockX;
-		int blockY;
-		if (pArr->x < 8) {
-			blockX = pArr->x;
-		} else {
-			blockX = 8;
-		}
-
-		if (pArr->y < 16) {
-			blockY = pArr->y;
-		} else {
-			blockY = 16;
-		}
-
-		for (xsec=0; xsec < pArr->x / blockX; xsec++) {
-			for (int ysec=0; ysec < pArr->y / blockY; ysec++) {
-				/* OK, we have a sector. Count mons, and wanted */
+		for (int xsec=0; xsec < pArr->x; xsec+=8) {
+			for (int ysec=0; ysec < pArr->y; ysec+=16) {
 				int mons=0;
 				int wanted=0;
-				for (int x=0; x<blockX; x++) {
-					for (int y=0; y<blockY; y+=2) {
-						ARegion *reg = pArr->GetRegion(x+xsec*blockX, y+ysec*blockY+x%2);
-						if (reg && !reg->IsGuarded()) {
+
+				for (int x=0; x < 8; x++) {
+					for (int y=0; y < 16; y+=2) {
+						ARegion *reg = pArr->GetRegion(x+xsec*8, y+ysec*16+x%2);
```

Comment:
You don't need to multiply xsec and ysec by 8 and 16 here, as this has already been done by the power of addition!

...however, you do need to check that x+sec < pArr->x and that y+ysec < pArr->y.  If you don't, it will still work, but will potentially skew monster spawn locations a bit.

Reply:
heh, missed that, thanks!